MantidQt/CustomInterfaces/test/ALCLatestFileFinderTest.h

```diff
@@ -16,36 +16,67 @@ using ScopedFileHelper::ScopedFile;
 using Mantid::Kernel::DateAndTime;
 
 /**
+ * Temporary directory that is deleted when it goes out of scope
+ */
+class ScopedDirectory {
+public:
+  /// Constructor: create directory in temp folder
+  ScopedDirectory(const std::string &dirName) : m_dirName(dirName) {
+    Poco::Path tmpPath(Mantid::Kernel::ConfigService::Instance().getTempDir());
+    tmpPath.pushDirectory(m_dirName);
+    m_directory = Poco::File(tmpPath);
+    m_directory.createDirectories();
+  }
+  /// Destructor: delete the directory
+  virtual ~ScopedDirectory() {
+    constexpr bool recursiveRemove(true);
+    m_directory.remove(recursiveRemove);
+  }
+  /// Get path of directory
+  std::string getDirectoryName() const { return m_dirName; }
+
+private:
+  Poco::File m_directory;
+  const std::string m_dirName;
```

Comment:
Are there downsides to having a const member variable? In particular, I'm concerned about copy and move assignment.